pkg/util/kubeobjects/internal/query/query.go

```diff
@@ -218,16 +218,22 @@ func (c Generic[T, L]) createOrUpdateForNamespaces(ctx context.Context, object T
 	return goerrors.Join(errs...)
 }
 
+func (c Generic[T, L]) DeleteForNamespace(ctx context.Context, objectName string, namespace string) error {
+	c.Log.Info("deleting object from namespace", "name", objectName, "namespace", namespace)
+
+	c.Target.SetName(objectName)
+	c.Target.SetNamespace(namespace)
+
+	return c.Delete(ctx, c.Target)
+}
```

Comment:
I like it, but IMO what we should do (in a follow up, as it will touch many things), is to replace the `Delete` with this `DeleteForNamespace`, as it's much nicer to use

Reasoning:
- `c.Delete` is annoying to use, because you have to create a whole object, eventhough the only thing that matters is the namespace and name, all the rest is useless info
- "Unpacking" an object is far simpler then creating a dummy object
   - what we have now:
      - when no object was built: `query.Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: GetSecretName(r.dk.Name), Namespace: r.dk.Namespace}})` 🤢 
      - when an object was built: `query.Delete(ctx, secret)` (this is RARE)
   - what we should have: 
     - when no object was built:`query.Delete(ctx, someName, someNamespace)` 	
     - when an object was built:`query.Delete(ctx, secret.Name, secret.Namespace)`

Could you create a follow up for this? 🥺